.cirrus.yml

```diff
@@ -0,0 +1,161 @@
+build_and_store_wheels: &BUILD_AND_STORE_WHEELS
+  install_cibuildwheel_script:
+    - python -m pip install cibuildwheel==2.10.1
+  cibuildwheel_script:
+    - cibuildwheel
+  wheels_artifacts:
+    path: "wheelhouse/*"
+
+
+######################################################################
+# Should wheels be built?
+# Test if the run was triggered by:
+# - a cron job. The cron job is not set in this file, but on the
+#   cirrus-ci repo page
+# - commit message containing [wheel build]
+# - a tag that begins with v, but doesn't end with dev0
```

Comment:
Probably harmless, but I'm pretty sure our release workflow suggests making sure the wheels build successfully before pushing the tag up, for fairly obvious reasons. If I continue to follow that approach on maintenance branches it would look like:

- push candidate release commit with `[wheel build]` in message
- if those wheel builds succeed, upload them to PyPI/GitHub release assets
- then push the tag (from local-only to public/shared)

So, it should be safe to assume that if I push the tag up, but it is associated with a commit that already has run through with `[wheel build]`, we won't rebuild the wheels, right?

Reply:
When a commit with [wheel build] is used the  wheels are made available as artifacts on the CI, nothing else. If built during a cron job or a manual trigger they're uploaded to anaconda nightly wheels. If they're built during a push and tag event the wheels are uploaded to anaconda staging.
We've not tested the last point yet, this may be the first time!

All wheels go through the test suite except macosx_arm64 and linux_aarch64. If they don't pass I think that wheel isn't uploaded (but have not checked that).

I would manually  start the wheel build and check that they're uploaded to anaconda nightly. Check that all of them have been made. If that occurs then I would make a tag and push that. The wheel build then goes again, with the wheels uploaded to staging. You then have the choice of whether to use those, or the ones manually triggered.

I don't know how the whole process will go on a maintenance branch.

I don't think we should backport this PR or 17029 until after the trigger logic has been sorted. Otherwise the cirrusci run will always be red unless it's making wheels.